### PR TITLE
Scarb docs, contracts -> contract in template, remove `lib-path` from cairo1

### LINKED ISSUE #1924: Improve the tab info for docs
I would suggest editing the page title in website/src/pages/index.tsx to start with "Protostar", now it start with "Cairo" and with a lot of tabs open it is usually the only visible word

Screenshot 2023-05-16 at 01 52 40

_Originally posted by @piotmag769 in https://github.com/software-mansion/protostar/issues/1910#issuecomment-1548767739_

### LINKED ISSUE #1957: Scarb docs and error messages
TODO:
- [x] Link scarb docs to warning and error messages (tagged with `TODO #1957`)

Update docs:
- [x] installation section - instructions how to install Scarb
- [x] dependencies management
- [x] everywhere where `cairo_project.toml` is mentioned
- [x] linked libraries

### LINKED ISSUE #1978: Docs refinement
TODO:
- [x] Missing articles, commas, grammar errors, typos etc.
- [x] Unify naming: Cairo 0, Cairo 1, Protostar, Scarb, Starknet etc.

### PR DESCRIPTION
**Pull request by piotmag769**
### Checklist
- [x] Relevant issue is linked
- [x] Docs updated/issue for docs created
- [ ] Added relevant tests

### CODE REVIEW

File: website/docs/cairo-1/02-understanding-cairo-packages.md [hunk 1]
```diff
@@ -78,57 +82,59 @@ fn returns_three() -> felt252 {
...
 
-#### Using added module
+#### Using the added module
+
+You can now use your function with `my_package::mod1::functions::returns_three()`.
 
-You now use your function with `hello_starknet::mod1::functions::returns_three()`.
+:::info
+The name `my_package` is the value of the `name` key in the `[package]` section of your `Scarb.toml`.
+:::
```
Comment (MaksymilianDemitraszek) [inline website/docs/cairo-1/02-understanding-cairo-packages.md:133]: I would merge this with Dependencies section adding a subsection here

File: website/docs/cairo-1/02-understanding-cairo-packages.md
Comment (MaksymilianDemitraszek) [inline website/docs/cairo-1/02-understanding-cairo-packages.md:1]: I would move this section after initialization section

File: website/docs/cairo-1/03-project-initialization.md [hunk 2]
```diff
@@ -9,45 +9,52 @@ sidebar_label: Project initialization
...
-
-### `hello_starknet`
-
-This directory contains our only package in this project - `hello_starknet`.
-
-### `cairo_project.toml` and `lib.cairo`
+### `Scarb.toml` and `lib.cairo`
 
-All Cairo1 packages must define these files.
+All Cairo 1 packages must define these files.
```
Comment (MaksymilianDemitraszek) [inline website/docs/cairo-1/03-project-initialization.md:36]: Those files define a cario package, you can learn more about those here.

File: website/docs/cairo-1/03-project-initialization.md [hunk 3]
```diff
@@ -9,45 +9,52 @@ sidebar_label: Project initialization
...
+[package]
+name = "hello_starknet"
+version = "0.1.0"
+
+[dependencies]
+```
+:::info 
+The name of the package is always the value
+of the `name` key in the `[package]` section of your `Scarb.toml`. 
+:::
```
Comment (MaksymilianDemitraszek) [inline website/docs/cairo-1/03-project-initialization.md:52]:
```suggestion
You can learn about [packages](./02-understanding-cairo-packages.md) and how
to [add new module to a package](./02-understanding-cairo-packages.md#adding-a-new-module) in
further sections.

```toml title="Scarb.toml"
```

File: website/docs/cairo-1/03-project-initialization.md [hunk 4]
```diff
@@ -9,45 +9,52 @@ sidebar_label: Project initialization
...
+
+### `src`
+
+This directory contains the source code of the package named `hello_starknet`. 
+
```
Comment (MaksymilianDemitraszek) [inline website/docs/cairo-1/03-project-initialization.md:57]:
```suggestion
```
Not very usefull, provides unnecessary clutter

**Review by MaksymilianDemitraszek — CHANGES_REQUESTED** [review]
This PR consists of three diffrent modifications. Each of them should be a separate PR. please split this one into three PRs, it is not a good practice to batch work like this 

- Scarb docs
- Renaming templates
- Removeing lib-path

File: website/docs/cairo-1/03-project-initialization.md [hunk 5]
```diff
@@ -90,60 +105,69 @@ the `protostar build` command and other commands to fail.
 
 ### Multi-contract project structure
```
Comment (MaksymilianDemitraszek) [inline website/docs/cairo-1/03-project-initialization.md:106]: This should be moved to packages section

File: website/docs/cairo-1/03-project-initialization.md
Comment (MaksymilianDemitraszek) [inline website/docs/cairo-1/03-project-initialization.md:1]: I feel like this section role should be somewhat a table of contents refering to relevant sections

File: website/docs/cairo-1/03-project-initialization.md [hunk 6]
```diff
@@ -90,60 +105,69 @@ the `protostar build` command and other commands to fail.
...
-package2 = "src"
-```
-
-Define each contract in the `[contracts]` section of the protostar.toml and each package
-in the `linked-librares`
+:::info 
+Even though `package1` and `package2` **directories** are inside `my_project` **directory**
+it does not make `package1` and `package2` **packages** parts of `my_project` **package**. 
+Therefore, you should refer to them using `package1::` and `package2::`.
+:::
```
Comment (MaksymilianDemitraszek) [inline website/docs/cairo-1/03-project-initialization.md:146]:
```suggestion
:::info 
:::
I don't think this is a correct abstraction. Everything imported to lib and its dependencies are parts of the package

```

Comment (piotmag769) [reply website/docs/cairo-1/03-project-initialization.md:146]: Yup, but those packages are not imported to lib, that's the point

File: website/docs/cairo-1/02-understanding-cairo-packages.md [hunk 7]
```diff
@@ -3,7 +3,10 @@
 There are several requirements that Cairo packages have to follow. These are explained in the following sections.
 
 You can refer to [official Cairo documentation](https://github.com/starkware-libs/cairo/tree/main/docs/reference) for
-more details.
+more details. 
+
+Keep in mind that Protostar does not support `cairo_project.toml`. 
+It uses [Scarb](https://docs.swmansion.com/scarb) - a powerful package manager -  and its [manifest files](https://docs.swmansion.com/scarb/docs/reference/manifest) instead.
```
Comment (mkaput) [inline website/docs/cairo-1/02-understanding-cairo-packages.md:9]:
```suggestion
It uses [Scarb](https://docs.swmansion.com/scarb) and its [manifest files](https://docs.swmansion.com/scarb/docs/reference/manifest) instead.
```

that part was non-technical and superfluous

File: website/docs/cairo-1/02-understanding-cairo-packages.md [hunk 8]
```diff
@@ -36,40 +39,41 @@ mod my_module {
...
 term "crates" for packages.
 :::
 
-### `cairo_project.toml`
+### `Scarb.toml`
 
-It is a required part of the package definition. It contains `[crate_roots]` parameter, which is the path to
-the `lib.cairo` file.
+It is a [Scarb's manifest files](https://docs.swmansion.com/scarb/docs/reference/manifest) used by Protostar to manage your dependencies with the help of Scarb.
+This way you can manage your [dependencies](https://docs.swmansion.com/scarb/docs/reference/specifying-dependencies) as if you were using Scarb directly - you will not notice any difference!
```
[the remark this answers is not in the document]
Comment (mkaput) [reply website/docs/cairo-1/02-understanding-cairo-packages.md:54]: I do actually find this last sentence completely unnecessary here. It's non-technical

File: website/docs/cairo-1/02-understanding-cairo-packages.md [hunk 9]
```diff
@@ -36,40 +39,42 @@ mod my_module {
...
-```toml title="cairo_project.toml"
-[crate_roots]
-my_package = "src"
+[dependencies]
 ```
 
 ### `lib.cairo`
 
-It is the root of the package tree. Here you can define functions, declare used modules, etc.
+It is the root of the package tree and ***must*** be placed inside `src` folder. Here you can define functions, declare used modules, etc.
```
Comment (mkaput) [inline website/docs/cairo-1/02-understanding-cairo-packages.md:68]:
```suggestion
It is the root of the package tree and it **must** be placed inside `src` folder. Here you can define functions, declare used modules, etc.
```

**Review by mkaput — APPROVED** [review]

File: website/docs/cairo-1/02-understanding-cairo-packages.md (hunk 7 above)
Comment (piotmag769) [reply website/docs/cairo-1/02-understanding-cairo-packages.md:9]: Yea, those are more (intended) advertisements of Scarb, you are right xD

File: website/docs/cairo-1/03-project-initialization.md (hunk 6 above)
Comment (MaksymilianDemitraszek) [reply website/docs/cairo-1/03-project-initialization.md:146]: Yup but those are dependencies of those imported to lib

File: website/docs/cairo-1/03-project-initialization.md (hunk 3 above)
Comment (piotmag769) [reply website/docs/cairo-1/03-project-initialization.md:52]: ?

File: website/docs/cairo-1/03-project-initialization.md
Comment (piotmag769) [reply website/docs/cairo-1/03-project-initialization.md:1]: I don't like this section either, but I don't really have any idea how to improve it without just repeating stuff that is in the next sections :/

**Review by MaksymilianDemitraszek — APPROVED** [review]
Do we have any tests for it /protostar/docs/cairo-1/understanding-cairo-packages#multi-contract-project-structure ?

**Comment by piotmag769** [comment]
> Do we have any tests for it /protostar/docs/cairo-1/understanding-cairo-packages#multi-contract-project-structure ?
Maybe not for this particular one, but in general: yea
https://github.com/software-mansion/protostar/tree/master/tests/e2e/fixtures/scarb_integration/multiple_contracts

**Pull request merged**